jobs/integr8ly/ocp3/nightly/osd-testing-nightly-trigger.yaml

```diff
@@ -0,0 +1,206 @@
+                                string(name: 'clusterName', value: "${CLUSTER_NAME}"),
+                                string(name: 'uninstallationGitBranch', value: "${INSTALLATION_BRANCH}"),
+                                string(name: 'routerShard', value: "${ROUTER_SHARD}"),
+                                string(name: 'openShiftToken', value: "${OPENSHIFT_TOKEN}"),
+                                string(name: 'userName', value: "${USERNAME}")
+                            ]).result
+                        }
+                    } // stage
+
+                    stage ('Run installation pipeline') {
+                        withCredentials([
+                                [$class: 'AmazonWebServicesCredentialsBinding', credentialsId: "${CREDENTIALS_ID}", accessKeyVariable: 'AWS_ACCESS_KEY_ID', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                                [$class: 'StringBinding', credentialsId: "rhmi-qe1-sso-client-secret", variable: 'SSO_SECRET'],
+                                [$class: 'StringBinding', credentialsId: "rhmi-qe1-token", variable: 'OPENSHIFT_TOKEN']
+                            ]) {
+                            installationStatus = build(job: 'osd-cluster-integreatly-install', propagate: false, parameters: [
+                                string(name: 'clusterName', value: "${CLUSTER_NAME}"),
+                                string(name: 'installationGitBranch', value: "${INSTALLATION_BRANCH}"),
+                                string(name: 'routerShard', value: "${ROUTER_SHARD}"),
+                                string(name: 'openShiftToken', value: "${OPENSHIFT_TOKEN}"),
+                                string(name: 'githubClientId', value: "${GH_CLIENT_ID}"),
+                                string(name: 'githubClientSecret', value: "${GH_CLIENT_SECRET}"),
+                                string(name: 'customerAdminPassword', value: "${CUSTOMER_ADMIN_PASSWORD}"),
+                                string(name: 'threeScaleAwsAccessKey', value: "${AWS_ACCESS_KEY_ID}"),
+                                string(name: 'threeScaleAwsSecretKey', value: "${AWS_SECRET_ACCESS_KEY}"),
+                                string(name: 'threeScaleBucketName', value: "${THREESCALE_BUCKET_NAME}"),
+                                string(name: 'threeScaleBucketRegion', value: "${THREESCALE_BUCKET_REGION}"),
+                                string(name: 'backupAwsKey', value: "${AWS_ACCESS_KEY_ID}"),
+                                string(name: 'backupAwsSecret', value: "${AWS_SECRET_ACCESS_KEY}"),
+                                string(name: 'backupsBucketName', value: "${BACKUPS_BUCKET_NAME}"),
+                                string(name: 'clientSecret', value: "${SSO_SECRET}")
+                            ]).result
+                        }
+                    } // stage
+
+                    if (installationStatus != 'SUCCESS' || uninstallationStatus != 'SUCCESS') {
```

Comment:
Does it make sense to proceed with the installation if uninstall fails? Why not to just fail right after the uninstall failure?

Reply:
Well it probably doesn't. I am always not sure how the try/catch works in such case. I will split up this if - one after uninstallation and second after - installation